### PR TITLE
Migrate DDL sub-classes in spark-extensions to JUnit5 and AssertJ style

### PR DESCRIPTION
**Pull request by tomtongue**
Migrate sub-classes in spark-extension to JUnit 5 along with https://github.com/apache/iceberg/pull/9613, and https://github.com/apache/iceberg/issues/9086.

For now, only one test `TestAlterTablePartitionFields` is updated with JUnit 5. Will add ~all~ DDL sub-classes in this PR.

### Current progress
- [x] TestAlterTablePartitionsFields
- [x] TestAlterTableSchema
- [x] TestBranchDDL
- [x] TestReplaceBranch
- [x] TestRequiredDistributionAndOrdering
- [x] TestSetWriteDistributionAndOrdering
- [x] TestTagDDL

TestViews will be separately created because of the code size (~ 1400 lines).

### CODE REVIEW

**Comment by nastra** [comment]
> Will add all sub-classes in this PR

depending on the size of the diff it's also fine to split this into 2-3 PRs. You could probably start within a specific package and combine subclasses within one or more packages together in a PR (again depending on the size of the diff).

**Comment by tomtongue** [comment]
Sure, thank you for the suggestion. The size of diff would be a bit big if I gather all the diffs in `extension` package in this PR. So, I will separately create PRs, but first add relevant classes (like DDLs) into this PR. If I misunderstand, please correct me.

**Comment by tomtongue** [comment]
@nastra Change the following 7 DDL extensions to JUnit 5 and AssertJ style. 

For now `Assertions.assertThatThrownBy` is not changed, and not statically imported (`assertThat` is statically imported like other classes).

If I need to add more classes in this PR, please let me know.

File: spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestReplaceBranch.java [hunk 1]
```diff
@@ -149,9 +148,6 @@ public void testReplaceBranchWithRetain() throws NoSuchTableException {
     long first = table.currentSnapshot().snapshotId();
     String branchName = "b1";
     table.manageSnapshots().createBranch(branchName, first).commit();
-    SnapshotRef b1 = table.refs().get(branchName);
-    Integer minSnapshotsToKeep = b1.minSnapshotsToKeep();
-    Long maxSnapshotAgeMs = b1.maxSnapshotAgeMs();
```
Comment (nastra) [inline spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestReplaceBranch.java:154]: why are these being removed? I think those need to stay

File: spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestSetWriteDistributionAndOrdering.java [hunk 2]
```diff
@@ -19,31 +19,30 @@
...
 import org.apache.iceberg.NullOrder;
+import org.apache.iceberg.ParameterizedTestExtension;
 import org.apache.iceberg.SortOrder;
 import org.apache.iceberg.Table;
 import org.apache.iceberg.TableProperties;
 import org.apache.iceberg.exceptions.ValidationException;
 import org.apache.spark.sql.internal.SQLConf;
 import org.assertj.core.api.Assertions;
-import org.junit.After;
 import org.junit.Assert;
```
Comment (nastra) [inline spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestSetWriteDistributionAndOrdering.java:32]: should be removed as well

**Review by nastra — COMMENTED** [review]
this looks almost ready to go, just 2 comments. Also the diff size is about right for reviewing.

File: spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestReplaceBranch.java (hunk 1 above)
Comment (tomtongue) [reply spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestReplaceBranch.java:154]: The sql in this test is like `ALTER TABLE catalog.db.tbl REPLACE BRANCH branch AS OF VERSION snapshotId RETAIN maxRefAge timeUnit`. For this query, the `minSnapshotsToKeep` and `maxSnapshotAgeMs` are always null in the current impl. If the query has `WITH SNAPSHOT RETENTION` the parameters are filled. So I remove this part and change the tests to null checks.

Comment (nastra) [reply spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestReplaceBranch.java:154]: ok thanks for confirming, no need to change anything as I think being explicit and making sure those are null is good in this case

Comment (tomtongue) [reply spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestReplaceBranch.java:154]: thank you!

**Review by nastra — APPROVED** [review]

**Comment by tomtongue** [comment]
Thanks so much for the review. Will create a PR for other sub-classes.

**Pull request merged**